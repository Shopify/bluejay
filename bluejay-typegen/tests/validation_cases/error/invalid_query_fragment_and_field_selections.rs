#[bluejay_typegen::typegen([
    type Query {
        foo: String
    }
])]
mod schema {
    #[query([
        {
            foo
            ...MyFragment
        }

        fragment MyFragment on Query {
            foo
        }
    ])]
    mod query {}
}

fn main() {}
