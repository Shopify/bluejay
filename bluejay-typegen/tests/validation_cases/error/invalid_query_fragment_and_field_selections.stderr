error: Error: typegen requires a fragment spread to be the only selection in the set
          ╭─[ <unknown>:1:1 ]
          │
        1 │ { foo ...MyFragment } fragment MyFragment on Query { foo }
          │ ──────────┬─┬────────
          │           ╰──────────── Selection set contains a fragment spread and other selections
          │             │
          │             ╰────────── Fragment spread
       ───╯
  --> tests/validation_cases/error/invalid_query_fragment_and_field_selections.rs:7:13
   |
7  |       #[query([
   |  _____________^
8  | |         {
9  | |             foo
10 | |             ...MyFragment
...  |
16 | |     ])]
   | |_____^
